metrics-operator/api/v1alpha3/analysis_types.go

```diff
@@ -24,23 +24,47 @@ import (
 type AnalysisSpec struct {
 	//Timeframe specifies the range for the corresponding query in the AnalysisValueTemplate
 	Timeframe `json:"timeframe"`
-	// Args corresponds to a map of key/value pairs that can be used to substitute placeholders in the AnalysisValueTemplate query. The placeholder must be the capitalized version of the key; i.e. for args foo:bar the query could be "query:percentile(95)?scope=tag(my_foo_label:{{.Foo}})".
+	// Args corresponds to a map of key/value pairs that can be used to substitute placeholders in the AnalysisValueTemplate query. i.e. for args foo:bar the query could be "query:percentile(95)?scope=tag(my_foo_label:{{.foo}})".
 	Args map[string]string `json:"args,omitempty"`
 	// AnalysisDefinition refers to the AnalysisDefinition, a CRD that stores the AnalysisValuesTemplates
 	AnalysisDefinition ObjectReference `json:"analysisDefinition"`
 }
 
+// ProviderResult stores reference of already collected provider query associated to its objective template
+type ProviderResult struct {
+	// Objective store reference to corresponding objective template
+	Objective ObjectReference `json:"objectiveReference,omitempty"`
+	// Value is the value the provider returned
+	Value string `json:"value,omitempty"`
+	// Err stores any possible error at retrieval time
+	Err string `json:"error,omitempty"`
```

Comment:
shall we rename this to something like `ErrMsg`, and call the json property `errMsg`? Otherwise, there will be a mismatch in the docs when looking at the description. Also, calling it `Err` might suggest that this is an `error` object, which is not the case here